operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/TimerEventSource.java

```diff
@@ -3,40 +3,53 @@
 import io.fabric8.kubernetes.client.CustomResource;
 import io.javaoperatorsdk.operator.processing.KubernetesResourceUtils;
 import io.javaoperatorsdk.operator.processing.event.AbstractEventSource;
+import java.io.IOException;
 import java.util.Map;
 import java.util.Timer;
 import java.util.TimerTask;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class TimerEventSource extends AbstractEventSource {
 
-  private Logger log = LoggerFactory.getLogger(TimerEventSource.class);
-
   private final Timer timer = new Timer();
-
+  private final AtomicBoolean running = new AtomicBoolean();
   private final Map<String, EventProducerTimeTask> onceTasks = new ConcurrentHashMap<>();
   private final Map<String, EventProducerTimeTask> timerTasks = new ConcurrentHashMap<>();
+  private Logger log = LoggerFactory.getLogger(TimerEventSource.class);
+
+  public boolean schedule(CustomResource customResource, long delay, long period) {
+    if (!running.get()) {
+      return false;
+    }
 
-  public void schedule(CustomResource customResource, long delay, long period) {
     String resourceUid = KubernetesResourceUtils.getUID(customResource);
     if (timerTasks.containsKey(resourceUid)) {
-      return;
+      return false;
     }
     EventProducerTimeTask task = new EventProducerTimeTask(resourceUid);
     timerTasks.put(resourceUid, task);
     timer.schedule(task, delay, period);
+
+    return true;
   }
 
-  public void scheduleOnce(CustomResource customResource, long delay) {
+  public boolean scheduleOnce(CustomResource customResource, long delay) {
```

Comment:
Is this really a case where the user should check the boolen value? Normally it should not happen that the controlerr is executed before start. So would rather throw a runtime exception in this case.

Reply:
I did it to make the signature consistent with `schedule(CustomResource,long,long period)` but I can do the same for both the methods

Reply:
@csviri migrated to exception